recipes/libsodium/1.0.18/conanfile.py

```diff
@@ -147,6 +149,11 @@ def _build_autotools(self):
             self._build_autotools_neutrino(configure_args)
         else:
             raise ConanInvalidConfiguration("Unsupported os for libsodium: {}".format(self.settings.os))
+            
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            if self.options.shared and "MT" in str(self.settings.compiler.runtime):
+                raise ConanInvalidConfiguration("Cannot build shared libsodium libraries with MT(d) runtime")
```

Comment:
```suggestion
        if self.options.shared and str(self.settings.compiler.get_safe("runtime")) in ["MT", "MTd", "static"]:
            raise ConanInvalidConfiguration("Cannot build shared libsodium libraries with MT(d) runtime")
```

More robust and handle clang-cl on Windows.

Reply:
Since it does not affect any consumer and it's a problem for all off CCI (except the one PR I saw you added this in) I'll pass since theres already two reviews.

Maybe this should be a hook?

Reply:
Ok, it can improved later